pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -640,15 +640,20 @@ private static String database(Properties props) {
    * @return the timeout from the URL, in milliseconds
    */
   private static long timeout(Properties props) {
+    int result = 0;
     String timeout = PGProperty.LOGIN_TIMEOUT.get(props);
     if (timeout != null) {
       try {
-        return (long) (Float.parseFloat(timeout) * 1000);
+        result = Integer.parseInt(timeout);
```

Comment:
Are there good reasons to convert from `float` to `int` parsing?
I agree it is a bit esoteric, however I would refrain from altering things without clear justification.